File: backend/app/schema/order.py
from marshmallow import Schema, fields
from typing import List

class OrderItemSchema(Schema):
    product_id = fields.Str()
    sku = fields.Str()
    name = fields.Str()
    quantity = fields.Int()
    price = fields.Float()

class OrderSchema(Schema):
    id = fields.Str()
    shopify_order_id = fields.Str()
    status = fields.Str()
    order_date = fields.DateTime()
    items = fields.List(fields.Nested(OrderItemSchema))
    customer_name = fields.Str()
    shipping_address = fields.Str()
    total_amount = fields.Float()

File: frontend/src/schema/order.ts
import { z } from 'zod';

// Interfaces
export interface OrderItem {
  product_id: string;
  sku: string;
  name: string;
  quantity: number;
  price: number;
}

export interface Order {
  id: string;
  shopify_order_id: string;
  status: string;
  order_date: Date;
  items: OrderItem[];
  customer_name: string;
  shipping_address: string;
  total_amount: number;
}

// HUMAN ASSISTANCE NEEDED
// The following schemas have a confidence level below 0.8 and may need review or adjustments
// Zod schemas
export const orderItemSchema = z.object({
  product_id: z.string(),
  sku: z.string(),
  name: z.string(),
  quantity: z.number().int().positive(),
  price: z.number().positive()
});

export const orderSchema = z.object({
  id: z.string(),
  shopify_order_id: z.string(),
  status: z.string(),
  order_date: z.date(),
  items: z.array(orderItemSchema),
  customer_name: z.string(),
  shipping_address: z.string(),
  total_amount: z.number().positive()
});